### PR TITLE
Aimbot Update

### PR DESCRIPTION
**Pull request by STEVE4git**
This contains the aimbot update from my fork
->200%+ improvement in projectile aimbot
->Huge gains for hitscan
->Autozoom fixed
-> Enjoy 
- STEVE4

### CODE REVIEW

File: src/hacks/Aimbot.cpp [hunk 1]
```diff
@@ -266,7 +270,7 @@ bool shouldbacktrack_cache = false;
 
 void updateShouldBacktrack()
 {
-    if (!hacks::tf2::backtrack::backtrackEnabled() || hacks::tf2::backtrack::hasData() || projectile_mode || !(*backtrackAimbot || force_backtrack_aimbot))
+    if (hacks::tf2::backtrack::hasData() || projectile_mode || !(*backtrackAimbot || force_backtrack_aimbot))
         shouldbacktrack_cache = false;
     else
         shouldbacktrack_cache = true;
```
Comment (BenCat07) [inline src/hacks/Aimbot.cpp:276]: Was this effect on backtrack tested?

File: src/hacks/Aimbot.cpp [hunk 2]
```diff
@@ -305,17 +309,17 @@ bool validateTickFOV(tf2::backtrack::BacktrackData &tick)
...
 }
 
-static void doAutoZoom(bool target_found)
+void doAutoZoom(bool target_found)
 {
     bool isIdle = target_found ? false : hacks::shared::followbot::isIdle();
 
     // Keep track of our zoom time
-    static Timer zoomTime{};
+    Timer zoomTime{};
```
Comment (BenCat07) [inline src/hacks/Aimbot.cpp:322]: AutoZoom worked, this is what broke it no?

File: src/hacks/Aimbot.cpp [hunk 3]
```diff
@@ -363,102 +367,94 @@ static void CreateMove()
...
+    
+    if (hacks::tf2::antianticheat::enabled)
+        fov = std::min(fov > 0.0f ? fov : FLT_MAX, 10.0f); 
+    bool should_backtrack = hacks::tf2::backtrack::backtrackEnabled();
+    int get_weapon_mode = g_pLocalPlayer->weapon_mode;
+    projectile_mode          = false;
+    projectileAimbotRequired = false;
+    bool should_zoom = *auto_zoom;
+    switch(get_weapon_mode){
+        case weapon_hitscan:{
```
Comment (BenCat07) [inline src/hacks/Aimbot.cpp:399]: Clang format this file please thanks

File: src/hacks/Aimbot.cpp [hunk 4]
```diff
@@ -1248,18 +1160,18 @@ void DoAutoshoot(CachedEntity *target_entity)
 }
 
 // Grab a vector for a specific ent
-Vector PredictEntity(CachedEntity *entity, bool vischeck)
+Vector PredictEntity(CachedEntity *entity)
 {
     // Pull out predicted data
     AimbotCalculatedData_s &cd = calculated_data_array[entity->m_IDX];
     Vector &result             = cd.aim_position;
-    if (cd.predict_tick == tickcount && cd.predict_type == vischeck && !shouldBacktrack(entity))
```
Comment (BenCat07) [inline src/hacks/Aimbot.cpp:1256]: Does this break backtrack aimbot?

File: src/hacks/Aimbot.cpp (hunk 2 above)
Comment (STEVE4git) [reply src/hacks/Aimbot.cpp:322]: I prefer autozoom like that. 
It autozooms if it sees an ent and continues autozooming after that (Really good for camping sniper spots). I think its also beneficial for bots who can make it to front lines quicker to wipe a team vs constantly zooming in and out.

File: src/hacks/Aimbot.cpp (hunk 1 above)
Comment (STEVE4git) [reply src/hacks/Aimbot.cpp:276]: Yes. Backtrack works fine from what I can tell

File: src/hacks/Aimbot.cpp (hunk 4 above)
Comment (STEVE4git) [reply src/hacks/Aimbot.cpp:1256]: Not from what I can tell.

File: src/hacks/Aimbot.cpp (hunk 2 above)
Comment (BenCat07) [reply src/hacks/Aimbot.cpp:322]: By making the timer non-static you effectively removed it

Comment (STEVE4git) [reply src/hacks/Aimbot.cpp]: Fixed

**Comment by uyukat** [comment]
why is my house on fire please help my grandma was in there i need your help now you're the only ones who can help me

**Pull request merged**